pkg/gitproviders/provider.go

```diff
@@ -675,3 +677,18 @@ func (n NormalizedRepoURL) Provider() GitProviderName {
 func (n NormalizedRepoURL) Protocol() RepositoryURLProtocol {
 	return n.protocol
 }
+
+func getOwnerFromUrl(url url.URL, providerName GitProviderName) (string, error) {
+	parts := strings.Split(url.Path, "/")
+	if len(parts) < 2 {
```

Comment:
Please write a test that passes in an empty url string and a string with only one "part".  Look at it from my perspective when I write a test, then write code, and then refactor (i.e. TDD).  I would pass in an empty and a second test with a partial url and it would actually panic because I don't have this if-statement in place.  I would then access the `parts[1]` and my code will panic because I don't have this check in place.  I would then add that check and both tests would pass.  We need to start capturing these sad path cases better and not just write happy path tests.

Reply:
I have added sad paths for empty url and missing owner. I also added a check because I learned you can have subgroups in subgroups in gitlab. That isnt currently supported because that can get messy the way we have it.